com.unity.shadergraph/Editor/Drawing/Controllers/ShaderInputViewController.cs

```diff
@@ -50,33 +50,43 @@ void ChangePropertyValue(GraphData graphData)
             {
                 case BooleanShaderProperty booleanProperty:
                     booleanProperty.value = ((ToggleData)newShaderInputValue).isOn;
+                    if (graphData.material) graphData.material.SetFloat(shaderInputReference.referenceName, booleanProperty.value ? 1.0f : 0.0f);
```

Comment:
How do we undo this if the user closes the graph, and doesn't save it?
We would have to reset the material back to defaults in that case, to keep it in sync with the actual state of the graph on disk.

Reply:
(and there's also the problem if unity crashes, it could leave the Material in a non-synced state... that wouldn't get cleaned up until you force-reimported the shadergraph)

Reply:
I tested the case with a crash and it works fine, because the generated material is never saved so when reopenning the project after a crash it will be in the state it was at the end of import

Reply:
nice!   Did you test doing "modify property default", Ctrl+S save project, then crash?

Reply:
I wonder if we can mark the material as "don't save this, runtime only modifications present" when we change it...

Reply:
Given the public API of the assetdatabase, even if we could do it internally I am pretty sure it's impossible for users to save changes to the material